fdbserver/DataDistribution.actor.cpp

```diff
@@ -2246,7 +2237,21 @@ ACTOR Future<Void> makeAuditProgressOnServer(Reference<DataDistributor> self,
 					audit->foundError = true;
 				} else {
 					ASSERT(phase == AuditPhase::Invalid);
-					AuditStorageRequest req(audit->coreState.id, auditState.range, audit->coreState.getType());
+					// Set doAuditOnStorageServer
+					if (self->remainingBudgetForAuditTasks.contains(auditType)) {
```

Comment:
Is it necessary to check it again here?

Reply:
Yes. Although we check it before spawning `makeAuditProgressOnServer`, a single makeAuditProgressOnServer may want to spawn multiple doAuditOnStorageServer's (for loop there). So, in the makeAuditProgressOnServer, we need to check before each spawn of doAuditOnStorageServer.

Reply:
When we are auditing storage server shard, do we need to divide `allKeys` into multiple sub-ranges and do them in parallel? Will a single task in DD and a single ACTOR in SS be sufficient? If for each storage server, we schedule only one task, we can check the `budget` only in `dispatchAuditStorageServerShard`.